File: Leetcode-one-by-one/src/Prime Number of Set Bits in Binary Representation.swift
class Solution {
    func countPrimeSetBits(_ left: Int, _ right: Int) -> Int {
        // Prime numbers possible for set bits (max ~32 for Int)
        let primes: Set<Int> = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31]
        
        var result = 0
        
        for num in left...right {
            let setBits = num.nonzeroBitCount
            if primes.contains(setBits) {
                result += 1
            }
        }
        
        return result
    }
}

File: Leetcode-one-by-one/src/Prime Number of Set Bits in Binary Representation.java
import java.util.*;
class Solution {
    public int countPrimeSetBits(int left, int right) {
        // Possible prime counts of set bits (max 32 bits for int)
        boolean[] isPrime = new boolean[33];
        int[] primes = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31};
        for (int p : primes) {
            isPrime[p] = true;
        }

        int count = 0;

        for (int i = left; i <= right; i++) {
            int setBits = Integer.bitCount(i);
            if (isPrime[setBits]) {
                count++;
            }
        }

        return count;
    }
}
